plan/physical_plan_builder.go

```diff
@@ -514,83 +580,41 @@ func (p *Join) convert2PhysicalPlan(prop *requiredProperty) (*physicalPlanInfo,
 	}
 	switch p.JoinType {
 	case SemiJoin, SemiJoinWithAux:
-		lChild := p.GetChildByIndex(0).(LogicalPlan)
-		rChild := p.GetChildByIndex(1).(LogicalPlan)
-		allLeft := true
-		for _, col := range prop.props {
-			if lChild.GetSchema().GetIndex(col.col) == -1 {
-				allLeft = false
-			}
-		}
-		join := &PhysicalHashSemiJoin{
-			WithAux:         SemiJoinWithAux == p.JoinType,
-			EqualConditions: p.EqualConditions,
-			LeftConditions:  p.LeftConditions,
-			RightConditions: p.RightConditions,
-			OtherConditions: p.OtherConditions,
-			Anti:            p.anti,
-		}
-		join.SetSchema(p.schema)
-		lProp := prop
-		if !allLeft {
-			lProp = &requiredProperty{}
-		}
-		if p.JoinType == SemiJoin {
-			lProp = removeLimit(lProp)
-		}
-		lInfo, err := lChild.convert2PhysicalPlan(lProp)
-		if err != nil {
-			return nil, errors.Trace(err)
-		}
-		rInfo, err := rChild.convert2PhysicalPlan(&requiredProperty{})
+		info, err = p.convert2PhysicalPlanSemi(prop)
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
-		resultInfo := join.matchProperty(prop, lInfo, rInfo)
-		if p.JoinType == SemiJoin {
-			resultInfo.count = uint64(float64(lInfo.count) * selectionFactor)
-		} else {
-			resultInfo.count = lInfo.count
-		}
-		if !allLeft {
-			resultInfo = enforceProperty(prop, resultInfo)
-		} else if p.JoinType == SemiJoin {
-			resultInfo = enforceProperty(limitProperty(prop.limit), resultInfo)
-		}
-		p.storePlanInfo(prop, resultInfo)
-		return resultInfo, nil
 	case LeftOuterJoin:
-		info, err := p.handleLeftJoin(prop, false)
+		info, err = p.convert2PhysicalPlanLeft(prop, false)
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
-		p.storePlanInfo(prop, info)
-		return info, nil
 	case RightOuterJoin:
-		info, err := p.handleRightJoin(prop, false)
+		info, err = p.convert2PhysicalPlanRight(prop, false)
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
-		p.storePlanInfo(prop, info)
-		return info, nil
 	default:
-		lInfo, err := p.handleLeftJoin(prop, true)
+		lInfo, err := p.convert2PhysicalPlanLeft(prop, true)
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
-		rInfo, err := p.handleRightJoin(prop, true)
+		rInfo, err := p.convert2PhysicalPlanRight(prop, true)
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
 		if rInfo.cost < lInfo.cost {
-			lInfo = rInfo
+			info = rInfo
+		} else {
+			info = lInfo
 		}
-		p.storePlanInfo(prop, lInfo)
-		return lInfo, nil
 	}
+	p.storePlanInfo(prop, info)
+	return info, nil
 }
 
-func (p *Aggregation) handleStreamAgg(prop *requiredProperty) (*physicalPlanInfo, error) {
+// convert2PhysicalPlanStream converts logical aggregation to stream aggregation *physicalPlanInfo.
```

Comment:
logical aggregation ->the logical aggregation
stream aggregation *physicalPlanInfo. -> the stream aggregation *physicalPlanInfo.